pkg/handlers/generic/lifecycle/registry/cncfdistribution/handler.go

```diff
@@ -59,14 +64,64 @@ func New(
 	}
 }
 
+// Setup ensures any pre-requisites for the CNCF Distribution registry addon are met.
+// It is expected to be called before the cluster is created.
+// Specifically, it ensures that the CA secret for the registry is created in the cluster's namespace.
+func (n *CNCFDistribution) Setup(
+	ctx context.Context,
+	_ v1alpha1.RegistryAddon,
+	cluster *clusterv1.Cluster,
+	log logr.Logger,
+) error {
+	log.Info("Setting up CA for CNCF Distribution registry")
+	err := utils.EnsureCASecretForCluster(
+		ctx,
+		n.client,
+		cluster,
+	)
+	if err != nil {
+		return fmt.Errorf("failed to ensure CA secret for CNCF Distribution registry addon: %w", err)
+	}
+	return nil
+}
+
+// Apply applies the CNCF Distribution registry addon to the cluster.
 func (n *CNCFDistribution) Apply(
 	ctx context.Context,
 	_ v1alpha1.RegistryAddon,
 	cluster *clusterv1.Cluster,
 	log logr.Logger,
 ) error {
-	log.Info("Applying CNCF Distribution registry installation")
+	// Copy the TLS secret to the remote cluster.
+	serviceIP, err := utils.ServiceIPForCluster(cluster)
+	if err != nil {
+		return fmt.Errorf("error getting service IP for the CNCF distribution registry: %w", err)
+	}
+	opts := &utils.EnsureCertificateOpts{
+		RemoteSecretKey: ctrlclient.ObjectKey{
+			Name:      tlsSecretName,
+			Namespace: DefaultHelmReleaseNamespace,
+		},
+		Spec: utils.CertificateSpec{
+			CommonName:  stsName,
+			DNSNames:    certificateDNSNames(),
+			IPAddresses: certificateIPAddresses(serviceIP),
+		},
+	}
+	err = utils.EnsureTLSCertificateSecretOnRemoteCluster(
```

Comment:
for separate PR: It would be good to provide a dev doc with diagram to demonstrate creation of global root CA, registry CA for a cluster and TLS certificates for a registry and how they will be copied, updated etc. 
Also good to document how client can push to this registry.

Reply:
Great idea, added a new section on cert handling.

I don't yet know how we want the push to happen since this registry is not exposed publicly so left that out.